shared/src/balancer/event_handler.rs

```diff
@@ -48,11 +49,11 @@ pub struct TokensRegistered {
 
 #[derive(Clone, Debug, Eq, PartialEq, Hash)]
 pub struct WeightedPool {
-    pool_id: H256,
-    pool_address: H160,
+    pub pool_id: H256,
+    pub pool_address: H160,
+    pub normalized_weights: Vec<U256>,
+    pub specialization: PoolSpecialization,
```

Comment:
Shouldn't the other fiels also be accessible?

Reply:
I suppose so, I was only trying to edit the parts that I needed for this PR and we will also find that soon this struct is only for internal use anyway.